index.js

```diff
@@ -16,14 +16,15 @@ const isCI =
 export const isColorSupported =
   !isDisabled && (isForced || isWindows || isCompatibleTerminal || isCI)
 
-const raw = (open, close, searchRegex, replaceValue) => (s) =>
-  (s += "")
-    ? open +
-      (~s.indexOf(close, 4) // skip opening \x1b[
-        ? s.replace(searchRegex, replaceValue)
-        : s) +
-      close
-    : ""
+const raw = (open, close, searchRegex, replaceValue) => 
+  (s = "") =>
+    ((s += "") === "")
+      ? s
+      : open +
+        (~s.indexOf(close, 4) // skip opening \x1b[
+          ? s.replace(searchRegex, replaceValue)
+          : s) +
+        close
```

Comment:
We don't need to `s+=""` eagerly. If we're in and `s` is `undefined` or `""` we can get out faster:

```js
const raw =
  (open, close, searchRegex, replaceValue) =>
  (s = "") =>
    s === ""
      ? s
      : open +
        (~(s += "").indexOf(close, 4) // skip opening \x1b[
          ? s.replace(searchRegex, replaceValue)
          : s) +
        close
```

Reply:
I tried that and had a faulty output on the empty array:

```node
> String([])
''
> c.blue([])
'\x1b[34m\x1b[39m'
```

Maybe putting a second, lazy comparison and print the empty string?

```suggestion
const raw = (open, close, searchRegex, replaceValue) => 
  (s = "") =>
    s === "" || s === []
      ? ""
      : open +
        (~(s += "").indexOf(close, 4) // skip opening \x1b[
          ? s.replace(searchRegex, replaceValue)
          : s) +
        close
```

Or, alternatively, we could just not care since the official typing doesn't expect an array here in the first place.

Reply:
If we did something with arrays, `s === []` would be fine, but we don't, so we don't really need it.

Reply:
@NickKaramoff

Reply:
```suggestion
const raw =
  (open, close, searchRegex, replaceValue) =>
  (s = "") =>
    s === ""
      ? s
      : open +
        (~(s += "").indexOf(close, 4) // skip opening \x1b[
          ? s.replace(searchRegex, replaceValue)
          : s) +
        close
```